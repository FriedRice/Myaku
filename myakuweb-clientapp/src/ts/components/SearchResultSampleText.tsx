/** @module Search result article sample text component */

import { ArticleSampleText } from '../types';
import React from 'react';

interface SearchResultSampleTextProps {
    sampleText: ArticleSampleText;
}
type Props = SearchResultSampleTextProps;


const SearchResultSampleText: React.FC<Props> = function(props) {
    var sampleSegments: React.ReactNode[] = [];
    var segments = props.sampleText.segments;
    for (let i = 0; i < segments.length; ++i) {
        if (segments[i].isQueryMatch) {
            sampleSegments.push(
                <span key={`segment-${i}`} className='article-sample-instance'>
                    {segments[i].text}
                </span>
            );
        } else {
            sampleSegments.push(segments[i].text);
        }
    }

    return (
        <blockquote className='article-sample-text'>
            <p className='japanese-text' lang='ja'>
                {sampleSegments}
            </p>
            <footer>
                {props.sampleText.textStartPos}
            </footer>
        </blockquote>
    );
};

export default SearchResultSampleText;
