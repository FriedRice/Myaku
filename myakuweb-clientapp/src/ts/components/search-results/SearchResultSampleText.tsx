/**
 * SearchResultSampleText component module. See [[SearchResultSampleText]].
 */

import { ArticleSampleText } from 'ts/types/types';
import React from 'react';

/** Props for the [[SearchResultSampleText]] component. */
interface SearchResultSampleTextProps {
    /** Sample text content to display in the component */
    sampleText: ArticleSampleText;
}
type Props = SearchResultSampleTextProps;


/**
 * Article search result sample text blockquote component.
 *
 * @param props - See [[SearchResultSampleTextProps]].
 */
const SearchResultSampleText: React.FC<Props> = function(props) {
    var sampleSegments: React.ReactNode[] = [];
    var segments = props.sampleText.segments;
    for (let i = 0; i < segments.length; ++i) {
        if (segments[i].isQueryMatch) {
            sampleSegments.push(
                <span key={`segment-${i}`} className='article-sample-instance'>
                    {segments[i].text}
                </span>
            );
        } else {
            sampleSegments.push(segments[i].text);
        }
    }

    return (
        <blockquote className='article-sample-text'>
            <p lang='ja'>
                {sampleSegments}
            </p>
            <footer>
                {props.sampleText.textStartPos}
            </footer>
        </blockquote>
    );
};

export default SearchResultSampleText;
