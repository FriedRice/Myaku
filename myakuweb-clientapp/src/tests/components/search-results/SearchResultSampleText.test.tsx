/**
 * Tests for the [[SearchResultSampleText]] component.
 */

import { BASELINE_SEARCH_RESULT_PAGE } from 'tests/testData';
import React from 'react';
import SearchResultSampleText from
    'ts/components/search-results/SearchResultSampleText';
import { shallow } from 'enzyme';


describe('<SearchResultSampleText />', function() {
    it('renders correctly', function() {
        const testResult = BASELINE_SEARCH_RESULT_PAGE.results[0];
        const wrapper = shallow(
            <SearchResultSampleText
                sampleText={testResult.mainSampleText}
            />
        );
        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({sampleText: testResult.moreSampleTexts[0]});
        expect(wrapper).toMatchSnapshot();
    });
});
