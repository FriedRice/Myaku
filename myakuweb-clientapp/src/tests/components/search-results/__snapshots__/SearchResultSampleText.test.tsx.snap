// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultSampleText /> renders correctly 1`] = `
<blockquote
  className="article-sample-text"
>
  <p
    lang="ja"
  >
    アメリカ
    <span
      className="article-sample-instance"
      key="segment-1"
    >
      の
    </span>
    一番
    <span
      className="article-sample-instance"
      key="segment-3"
    >
      の
    </span>
    ネコ
  </p>
  <footer>
    50% into article
  </footer>
</blockquote>
`;

exports[`<SearchResultSampleText /> renders correctly 2`] = `
<blockquote
  className="article-sample-text"
>
  <p
    lang="ja"
  >
    アメリカ
    <span
      className="article-sample-instance"
      key="segment-1"
    >
      の
    </span>
    心
  </p>
  <footer>
    30% into article
  </footer>
</blockquote>
`;
